### PR TITLE
document class instantiation for tests inside classes

### LINKED ISSUE #6900: Documented that each test gets their own instance of the class?
Is it documented anywhere that each test gets their own instance of the class?  I was a little shocked when I figured that out.

I also think it would make sense to change the behavior so that every test within a class shared the same instance of the class.

_Originally posted by @szegel in https://github.com/pytest-dev/pytest/issues/6890#issuecomment-597828312_

### PR DESCRIPTION
**Pull request by symonk**
brief documentation to outline that tests inside classes do not leak/persist/share the same class instance but instead, they create a new one.  closes #6900

### CODE REVIEW

File: doc/en/getting-started.rst [hunk 1]
```diff
@@ -153,6 +153,57 @@ Once you develop multiple tests, you may want to group them into a class. pytest
 
 The first test passed and the second failed. You can easily see the intermediate values in the assertion to help you understand the reason for the failure.
 
+Grouping tests in classes can be beneficial for the following reasons:
+
+ * Test organization
+ * Sharing fixtures for tests only in that particular class
+ * Applying marks at the class level and having them implicitly apply to all tests
+
+Something to be aware of when grouping tests inside classes is that each test does not have the same instance of the class.
```
Comment (RonnyPfannschmidt) [inline doc/en/getting-started.rst:162]:
```suggestion
Something to be aware of when grouping tests inside classes is that each test does have a own unique instance of the class.
```

Comment (symonk) [reply doc/en/getting-started.rst:162]: Thanks @RonnyPfannschmidt - thoughts on:

'each test has a unique instance of the class'.

a slight middle ground there?

Comment (RonnyPfannschmidt) [reply doc/en/getting-started.rst:162]: @symonk great idea, i just wanted to get away from "not the same" - your suggestion reads better :+1:

**Comment by symonk** [comment]
not sure whats up with coverage, its merely doc changes

**Comment by symonk** [comment]
Unable to merge, could someone with appropriate access do that when they have the chance :) appreciate it thanks

**Comment by RonnyPfannschmidt** [comment]
thanks for the nudge

**Comment by symonk** [comment]
@RonnyPfannschmidt thank you very much!

**Pull request merged**